lib/proxy.js

```diff
@@ -1,88 +1,135 @@
-'use strict';
-var cookie = require('simple-cookie');
-var httpProxy = require('http-proxy');
-var url = require('url');
+"use strict";
+var cookie = require("simple-cookie");
+var httpProxy = require("http-proxy");
+var url = require("url");
 
 function joinPath(p1, p2) {
-    return p1.replace(/\/?$/, '/') + p2.replace(/^\//, '');
+  return p1.replace(/\/?$/, "/") + p2.replace(/^\//, "");
+}
+
+function requestLogger(message, doLogging) {
+  if (!doLogging) {
+    return () => {};
+  }
+  return function(req) {
+    const headers = req.headers || req.getHeaders();
+    const headerStr = Object.keys(headers)
+      .map(header => {
+        return `  ${header}: ${headers[header]}`;
+      })
+      .sort()
+      .join("\n");
+
+    console.log(
+      `
+${message} ${headers.host}
+${req.method} ${req.path}
+${headerStr}
+`
+    );
+  };
+}
+
+function verboseLogDelimiter() {
+  console.log(`=======================================================`);
 }
 
 module.exports = exports = function genProxyHandler(server, options) {
-    var destinationUrl = options.destinationUrl;
-    var targetPath = options.targetPath;
-    var proxy = httpProxy.createProxyServer({ secure: !options.insecure });
-    var destUrl = url.parse(destinationUrl);
-    var targetRegExp = new RegExp('^' + targetPath);
-
-    function shouldHandle(req) {
-        var path = url.parse(req.url).path;
-        return targetRegExp.test(path);
-    }
+  var destinationUrl = options.destinationUrl;
+  var targetPath = options.targetPath;
+  var proxy = httpProxy.createProxyServer({
+    secure: !options.insecure,
+    changeOrigin: true,
+    hostRewrite: true,
+    autoRewrite: true,
+    prependPath: true,
+    ws: true
+  });
+  var destUrl = url.parse(destinationUrl);
+  var targetRegExp = new RegExp("^" + targetPath);
 
-    function cleanUpRequest(req) {
-        var reqUrl = url.parse(req.url);
-        reqUrl.pathname = reqUrl.pathname.replace(targetRegExp, '');
-        req.url = url.format(reqUrl);
+  var logInboundRequest = requestLogger("Inbound to", options.verbose);
+  var logOutboundRequest = requestLogger("Outbound to", options.verbose);
+  var logDelimiter = options.verbose ? verboseLogDelimiter : function() {};
```

Comment:
Added some verbose logging options in the laziest possible way, but now it's easier to see details to help debug proxied requests.